content/ember-4-5-released.md

```diff
@@ -0,0 +1,152 @@
+---
+title: Ember 4.5 Released
+authors:
+  - jen-weber
+date: 2022-06-20T00:00:00.000Z
+tags:
+  - releases
+  - '2022'
+  - version-4-x
+---
+
+Today the Ember project is releasing version 4.5 of Ember.js, Ember Data, and Ember CLI.
+
+This release kicks off the 4.5 beta cycle for all sub-projects. We encourage our community (especially addon authors) to help test these beta builds and report any bugs before they are published as a final release in six weeks' time. The [ember-try](https://github.com/ember-cli/ember-try) addon is a great way to continuously test your projects against the latest Ember releases.
+
+You can read more about our general release process here:
+
+- [Release Dashboard](http://emberjs.com/releases/)
+- [The Ember Release Cycle](https://blog.emberjs.com/new-ember-release-process/)
+- [The Ember Project](https://blog.emberjs.com/ember-project-at-2-0/)
+- [Ember LTS Releases](https://blog.emberjs.com/announcing-embers-first-lts/)
+
+---
+
+## Ember.js
+
+Ember.js is the core framework for building ambitious web applications.
+
+### Changes in Ember.js 4.5
+
+Ember.js 4.5 is an incremental, backwards compatible release of Ember with bug fixes, performance improvements, and minor deprecations.
+
+#### Bug Fixes
+
+Ember.js 4.5 introduced 0 bug fixes.
+
+#### Features
+
+Ember.js 4.5 introduced 2 new features.
+
+1. Plain function as helpers
+2. `renderSettled` is now public API
+
+##### 1. Plain functions as helpers
+
+You can now use plain functions as helpers in your component templates. This helps make the relationship between Ember component templates and their JavaScript class more intuitive.
+The full documentation can be found in [TODO link to guides]()
+
+For example, here we create a method `double` and use it directly in a template:
+
+```js
+// my-component.js
+
+import Component from '@glimmer/component';
+
+export default class MyComponent extends Component {
+  double = num => num * 2;
+}
+```
+
+```hbs
+// my-component.hbs
+
+{{this.double 2}}
+
+<SomeComponent @foo={{this.double 2}} />
+```
+
+Previously, you would need to [write a separate helper](https://guides.emberjs.com/release/components/helper-functions/#toc_writing-a-helper-function) in order to accomplish this.
+
+This feature was made possible when [RFC 756](https://rfcs.emberjs.com/id/0756-helper-default-manager) was implemented.
+
+##### 2. `renderSettled` is now public API
+
+`renderSettled` is a method you can use in your application tests. For example, in a test, you might want to update some tracked state and then run some assertions after rendering has completed. The `renderSettled` method returns a promise which fulfills when rendering has completed.
+
+```
+import { renderSettled } from '@ember/renderer';
+```
+
+`renderSettled` is one chapter in some upcoming improvements to Ember's testing story. The ongoing work and goals are described in [RFC 785](https://rfcs.emberjs.com/id/0785-remove-set-get-in-tests).
```

Comment:
````suggestion
Under the hood, Ember's tests use a "test waiters" system to allow you to control the flow of your tests in terms of the actual framework behavior.
That way that your tests match exactly what the app does at runtime.
However, making this work depends on Ember providing all the necessary hooks for test helpers to use, and there was one significant missing public API.
You could wait for *all* of the test waiters to finish with `settled`, but there was no public way to wait for *just rendering* to finish.
For example, you might want to wait for rendering to finish but *not* for an Ember Data `save` operation to finish, as part of testing a loading screen.

Ember 4.5 introduces a new function, `renderSettled`, as a public way for test helpers to interact with the rendering phase of the application.
`renderSettled` returns returns a promise which fulfills as soon as rendering has completed.
It can be used in any rendering or application test.
(It also works in other tests where you set up the rendering hooks manually, but this is unusual!)

```
import { renderSettled } from '@ember/renderer';
```

An upcoming release of `@ember/test-helpers` will take advantage of this to provide a new `await rerender()` helper.
For more details, and how this fits into improvements to Ember's testing story, see [RFC 785](https://rfcs.emberjs.com/id/0785-remove-set-get-in-tests).
````